src/shared/machine-id-setup.c

```diff
@@ -140,7 +141,7 @@ int machine_id_setup(const char *root, bool force_transient, sd_id128_t machine_
         }
 
         /* A we got a valid machine ID argument, that's what counts */
-        if (sd_id128_is_null(machine_id)) {
+        if (sd_id128_is_null(machine_id) || sd_id128_is_firmware(machine_id)) {
```

Comment:
please change machine_id_setup() so that it gets a flags param, and then define one flag that tells us to initialize the machine ID from firmware. Please fold the existing bool param `force_transient` into that new flags param.

specifically, change this:

```c
int machine_id_setup(const char *root, bool force_transient, sd_id128_t machine_id, sd_id128_t *ret);
```

to this:
```c
typedef enum MachineIdSetupFlags {
        MACHINE_ID_SETUP_FORCE_TRANSIENT = 1 << 0,
        MACHINE_ID_SETUP_FORCE_FIRMWARE  = 1 << 1, 
} MachineIdSetupFlags;

int machine_id_setup(const char *root, sd_id128_t machine_id, MachineIdSetupFlags flags, sd_id128_t *ret);
```